### DISCUSSION TITLE
Day 1 Advent of Coding Challenge

### DISCUSSION THREAD

**Discussion by CarolineMorton** [CarolineMorton/AdventOfCode_WIR#2] [Ideas]
⭐ **Day 1 Puzzle is out!** ⭐

Hello everyone!

The first puzzle is out and is available here: https://adventofcode.com/2024/day/1

Make sure you sign up to the puzzle via the above link. I did this via Github but there are a few options. 

How are you all getting on with Day 1?

We have the kick off meeting today where we can discuss this further. For now, I am getting puzzling! 

See you all later. 

Caroline

**Comment by LizzieH01** [comment]
Looking forward to this!

**Comment by AR2202** [comment]
Solved the first couple in Haskell as mentioned in the call. Repo is [here](https://github.com/AR2202/AOC2024) in case anyone is interested. Hope to get to Rust at some point.

**Reply by CarolineMorton** [reply]
Had a look at this and it looks great! I must attempt Haskell one day ...!

**Comment by hollygrimm** [comment]
[My solution](https://github.com/hollygrimm/AdventOfCode_WIR/tree/main/day_01) for Day 1 in Rust (updated with Part 2)

**Reply by CarolineMorton** [reply]
Love the error mapping using Enums here: https://github.com/hollygrimm/AdventOfCode_WIR/blob/b71aa35ff6b4cedfcea64b4ab1e1b5c5d6d251ee/day_01/src/main.rs#L34

**Comment by Sowmiya07** [comment]
[Here](https://github.com/Sowmiya07/AdventOfCode_WIR/blob/main/day_01/src/main.rs) is mine

**Comment by AR2202** [comment]
[day 1 in Rust](https://github.com/AR2202/AdventOfCode_WIR/tree/main/day_01)

**Comment by em-baggie** [comment]
Here's <a href = https://github.com/em-baggie/AdventOfCode_WIR/blob/main/day_01/src/main.rs>my solution</a> to day 1 (parts 1 and 2) in Rust. Sorry to miss the call earlier!

**Reply by CarolineMorton** [reply]
Great to see this @em-baggie

**Comment by DominikStefancik** [comment]
My [solution for the day 1](https://github.com/DominikStefancik/AdventOfCode_WIR/tree/main/day_01)

**Comment by CarolineMorton** [comment]
This is my solution: https://github.com/CarolineMorton/AdventOfCode_WIR/tree/caroline-solutions/day_01

On a branch in this repo so people can choose not to see if when they fork the repo. 

I chose to concentrate on using Command line tool for this puzzle as that's an area i want to improve. I used clap for this.

**Comment by fatim** [comment]
My version - [Day 1]( https://github.com/fatim/AdventOfCode_2024_Rust/blob/main/day_01/)

**Comment by CarolineMorton** [comment]
Does anyone need any help?

**Comment by kuhmil** [comment]
Fell a little behind, here is my version of [Day 1](https://github.com/kuhmil/AdventOfCode_WIR/tree/main/day_01/src)!

**Reply by CarolineMorton** [reply]
I think we are all behind - this happens every year!! 🎅🎅 I am choosing to think that taking part is half the battle!
